app/controllers/admins/sessions_controller.rb

```diff
@@ -0,0 +1,13 @@
+# frozen_string_literal: true
+
+class Admins::SessionsController < Devise::SessionsController
+  layout 'admins/application'
+
+  def after_sign_in_path_for(resource)
+    admins_root_path
+  end
```

Comment:
namespace を admin にしておくと定義不要かもしれない(after_sign_out_path_for の方も)
https://github.com/heartcombo/devise/blob/e2242a95f3bb2e68ec0e9a064238ff7af6429545/lib/devise/controllers/helpers.rb#L172-L175

あるいはこうするとよいかも
```suggestion
  def after_sign_in_path_for(resource)
    stored_location_for(resource) || admins_root_path
  end
```

Reply:
うーん、、、いまいちdeviseでnamespace区切ったときの挙動がわからなかったです…
ひとまず、`stored_location_for(resource) || `を追記していますが、現状試せるページがないので動作確認は今後実施します
[de6215a](https://github.com/kk0000-kk/sakura-market/pull/17/commits/de6215a343425429bf2f786999446c8bb30a28ed)

Reply:
こんな感じです。

```ruby
        scope = Devise::Mapping.find_scope!(resource_or_scope) # Adminモデルなら `:admin`
        router_name = Devise.mappings[scope].router_name       # この行はいったん気にしない

        home_path = "#{scope}_root_path"                       # `scope == :admin` なので `admin_root_path`
```